rj_gameplay/stp/eval/__init__.py

```diff
@@ -0,0 +1,197 @@
+        "DEFEND_PENALTY" : "Our opponents are performing a penalty kick",
+        "OFFENSIVE_KICK" : "We are performing a restart in the offensive area of the field",
+        "DEFENSIVE_KICK" : "We are performing a restart in the defensive area of the field",
+        "MIDFIELD_KICK" : "We are performing a restart in the midfield area of the field",
+        "GOALIE_CLEAR" : "Our goalie safely has the ball inside our goalie zone",
+        }
+
+    @property
+    def description(self) -> str:
+        return self.descriptions.get(self.name, default = "")
+
+    @property
+    def name_dec(self) -> str:
+        return self.fname + self.description
+
+    @property
+    def fname(self) -> str:
+        return string.capwords(self.name.lower())
+
+    def __str__(self):
+        return "Situation: " + self.fname
+
+    def is_clear(self):
+        pass
+
+    def is_scramble(self):
+        pass
+
+    def is_offensive(self):
+        pass
+
+    def is_midfield(self):
+        pass
+
+    def is_defensive(self):
+        pass
+
+PropT = TypeVar("PropT")
+
+class IEvaluator(Generic[PropT], ABC):
+    """
+    Current plan, evaluators tick on the old evaluation state and produce a new one.
+    Inside the PropT object evaluated information can be found
+    """
+
+    @abstractmethod
+    def tick(self, world_state: rc.WorldState, prev_props: Optional[PropT]) -> PropT:
+        """
+        Performs a tick of this evaluator
+        :param world_state: The state of the world
+        :param prev_props: The props from the previous tick if available
+        :return: new props
+        """
+        ...
+
+
+#class ISituationEvaluator(IEvaluator):
+#    """Interface for situation evaluator"""
+#
+#    @abstractmethod
+#    def get_situation(self, props: PropT) -> Situation:
+#        """
+#        :param props: The properties from a tick of the evaluator
+#        :return: The situation from the props
+#        """
+#        ...
+
+
+"""
+Situation Characteristics:
+Pressure
+Optimism/Advantage
+
+
+Match Characteristics:
+Exploration
+Bias (per play)
+
+Play Characteristics:
+Speed
+Reliability
+Reward
+Fit (dynamic)
+"""
+
+@dataclass
+class PlaySelectInfo():
+    """
+    The idea would be that these would be added in the playbook alongside the play for the play selector to use when determining which play to run. That would allow for the tuning of play 
+
+    This could be an interface, and playbooks and playselectors just have to provide some kind of play selection info along with the play list...
+    """
+
+    ##How long does this play need to execute in relation to other plays in the same situation (0, 1.0)
+    speed: float
+    ##How reliable do we consider this play (0,1)
+    reliability: float
+    ##???
+    reward: float
+    ##How well does the play think it fits within the situation
+    fit: float
+    ##Ad-Hoc bias term to increase or decrease the probability of a play running (-inf, inf)
```

Comment:
Are you going for a more probabilistic model for play selection, or just altering the cost by some random noise?